moveit_core/collision_detection/src/world.cpp

```diff
@@ -295,9 +363,45 @@ bool World::setSubframesOfObject(const std::string& object_id, const moveit::cor
     ASSERT_ISOMETRY(t.second)  // unsanitized input, could contain a non-isometry
   }
   obj_pair->second->subframe_poses_ = subframe_poses;
+
+  updateGlobalPoses_(obj_pair->second, false, true);
   return true;
 }
 
+bool World::setObjectPose(const std::string& object_id, const Eigen::Isometry3d& pose)
+{
+  ASSERT_ISOMETRY(pose);  // unsanitized input, could contain a non-isometry
+  ObjectPtr& obj = objects_[object_id];
+  int action = 0;
+  if (!obj)
+  {
+    obj.reset(new Object(object_id));
+    action |= CREATE;
+  }
+  ensureUnique(obj);
+  obj->pose_ = pose;
+  updateGlobalPoses_(obj);
+  notify(obj, Action(action));
+  return true;
+}
+
+void World::updateGlobalPoses_(ObjectPtr& obj, bool update_shape_poses, bool update_subframe_poses)
+{
+  // Update global shape poses
+  if (update_shape_poses)
+    for (unsigned int i = 0; i < obj->global_shape_poses_.size(); ++i)
+      obj->global_shape_poses_[i] = obj->pose_ * obj->shape_poses_[i];
```

Comment:
Is the size of `obj->global_shape_poses_` always smaller or the same as `obj->shape_poses_`?  If not this could cause problems.

Reply:
Yes, `global_shape_poses_` is only changed inside `world.cpp` and is only manipulated together with `shape_poses_`.